### ISSUE TITLE
Rule Change: Add fenced-code-meta rule

### ISSUE THREAD

**Issue by TKDev7** [eslint/markdown#477]
### What rule do you want to change?

fenced-code-language

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

\```js startline=3
const message = "Hello, world!";
console.log(message);
\```

### What does the rule currently do for this code?

Currently, the rule only validates the presence and allowance of the language identifier. It ignores any additional content in the info string (e.g., `startline=3`).

### What will the rule do after it's changed?

When `languageOnly` is `true`, the rule will report an error if the info string contains anything beyond the language identifier itself.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

This option aligns with markdownlint's [MD040](https://github.com/DavidAnson/markdownlint/blob/main/doc/md040.md) rule's `language_only` parameter, offering stricter control and promoting consistent rendering.

**Comment by snitin315** [comment]
`allowInfoString` makes sense 👍🏻

**Comment by lumirlumir** [comment]
Actually, I'm leaning toward using `checkMeta` as the option name for the following reasons:

- `infoString` -> `meta`: As @TKDev7 mentioned, `info` is a general term that includes code fence language and other meta information. Mdast uses `meta` for the property name we want to check now:

<img width="872" height="120" alt="Image" src="https://github.com/user-attachments/assets/8d9c7201-c6be-4700-aaf1-71a806021b09" />

<img width="1903" height="591" alt="Image" src="https://github.com/user-attachments/assets/ffa65e7d-8a92-4521-99df-fa7169fbfd7b" />

- Throughout the implementation of Markdown rules, the boolean option is usually named `check`. Here are some references:

https://github.com/eslint/markdown/blob/614ef3e0bf539655170f202467119fbbbf8963f6/src/rules/no-duplicate-headings.js#L64-L66

https://github.com/eslint/markdown/blob/614ef3e0bf539655170f202467119fbbbf8963f6/src/rules/no-empty-definitions.js#L73-L75

https://github.com/eslint/markdown/blob/614ef3e0bf539655170f202467119fbbbf8963f6/src/rules/table-column-count.js#L44-L46

**Comment by JoshuaKGoldberg** [comment]
+1 to having an option to do with `meta`, as that's often what it's called.

But - does this responsibility belong to `fenced-code-language`? The rule right now only pertains to language concerns, not additional meta(data). I'd think this would be best as a new rule.

I'm also not confident this option/rule makes sense to have on by default. Meta is pretty common in many Markdown scenarios, such as the docs generators used in Astro, Docusaurus, etc.

**Comment by nzakas** [comment]
I think @JoshuaKGoldberg's point makes sense. If what we want to do is disallow meta then that does seem outside the scope of this rule and perhaps a new rule is what we need.

**Comment by lumirlumir** [comment]
Is there any consensus on the approach we should take here? I think it would be helpful to have a clear direction moving forward.

**Comment by nzakas** [comment]
@SwetaTanwar we haven't yet decided the path forward here.

I think the approach should be to create a new rule. Perhaps we call it `fenced-code-meta` that can be set to "alway" or "never"?

Thoughts?

**Comment by lumirlumir** [comment]
+1 for creating a new `fenced-code-meta` rule.

I think if we add this rule now, it'll make it easier to extend its behavior regarding `meta` in the future.

**Comment by nzakas** [comment]
@TKDev7 I'd say it's safe to move forward.
